test/pylib/rest_client.py

```diff
@@ -95,9 +95,11 @@ async def post(self, resource_uri: str, host: Optional[str] = None,
                           json = json)
 
     async def put_json(self, resource_uri: str, data: Mapping, host: Optional[str] = None,
-                       port: Optional[int] = None, params: Optional[dict[str, str]] = None) -> None:
-        await self._fetch("PUT", resource_uri, host = host, port = port, params = params,
-                          json = data)
+                       port: Optional[int] = None, params: Optional[dict[str, str]] = None,
+                       response_type: Optional[str] = None) -> Any:
+        ret = await self._fetch("PUT", resource_uri, response_type = response_type, host = host,
```

Comment:
PUT does not return a result in REST. If it returns a result set it's either misnamed (should be named post) or misused (post should be used instead).

Reply:
Whatever the REST recommendations or specifications are... we're generally not using HTTP methods as we should be using them - e.g. we're often using GET for request that modify state. Using POST would make the PR bigger (I'd have to setup some scaffolding). Does it matter?

Reply:
The entire thing generally deserves refactor (using the proper methods, such as POST for state-changing operations and so on). We agreed to keep this out of this PR's scope.